Refactor non-standard date strings into a table (#38014)
mdn/content: files/en-us/web/javascript/reference/global_objects/date/parse/index.md

This function is useful for setting date values based on string values, for example in conjunction with the {{jsxref("Date/setTime", "setTime()")}} method.

> The spec doesn't explicitly specify what formats `parse()` can parse, but it does give a few invariants:

Reviewer: ```suggestion
The formats that `parse()` can handle are not explicitly specified, but there are a few invariants:
```

Date.parse("2019-01-01T00:00:00");
```

### toString() and toUTCString() formats

Apart from the standard date time string format, the {{jsxref("Date/toString", "toString()")}} and {{jsxref("Date/toUTCString", "toUTCString()")}} formats are supported:

```js
// toString() format
Date.parse("Thu Jan 01 1970 00:00:00 GMT-0500 (Eastern Standard Time)");
// 18000000 in all implementations in all timezones

// toUTCString() format
Date.parse("Thu, 01 Jan 1970 00:00:00 GMT");
// 0 in all implementations in all timezones
```

### Non-standard date strings

> [!NOTE]

> > This section contains implementation-specific behavior that can be inconsistent across implementations and across versions. It is not meant to be a comprehensive browser compatibility table and you should always conduct your own tests before using any format in your code.

Reviewer: ```suggestion
> This section contains implementation-specific behavior that may be inconsistent across browsers or different versions of browsers. It is not meant to be a comprehensive browser compatibility table and you should always conduct your own tests before using any format in your code.
```

Implementations usually default to the local time zone when the date string is non-standard. For consistency, we will assume that the runtime uses the UTC timezone, and unless specified otherwise, the output will vary with the device's time zone. [Daylight Saving Time (DST), of the local time zone, can also have an effect on this](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset#varied_results_in_daylight_saving_time_dst_regions).

> Here are some more examples of very non-standard date strings. Browsers parse date strings really generously and may decide to drop any part of the string that they can't parse, and they often copy each other's behavior for compatibility, so again, the goal here is only to provide some examples, and is not comprehensive in any way.

Reviewer: What do you think?

```suggestion
Here are some more examples of non-standard date strings. Browsers are very lenient when parsing date strings and may discard any part of a string that they cannot parse. For compatibility reasons, browsers often copy each other's behavior, so these handling patterns tend to propagate cross-browser. As previously stated, the following examples are for illustration only, and are not exhaustive by any means:
```
